lib/Map/MapboxVectorTileImageryProvider.js

```diff
@@ -0,0 +1,339 @@
+    }
+
+    this._errorEvent = new CesiumEvent();
+
+    this._ready = true;
+};
+
+defineProperties(MapboxVectorTileImageryProvider.prototype, {
+    url : {
+        get : function() {
+            return this._uriTemplate.expression;
+        }
+    },
+
+    tileWidth : {
+        get : function() {
+            return this._tileWidth;
+        }
+    },
+
+
+    tileHeight: {
+        get : function() {
+            return this._tileHeight;
+        }
+    },
+
+
+    maximumLevel : {
+        get : function() {
+            return this._maximumLevel;
+        }
+    },
+
+
+    minimumLevel : {
+        get : function() {
+            return this._minimumLevel;
+        }
+    },
+
+
+    tilingScheme : {
+        get : function() {
+            return this._tilingScheme;
+        }
+    },
+
+
+    rectangle : {
+        get : function() {
+            return this._rectangle;
+        }
+    },
+
+
+    errorEvent : {
+        get : function() {
+            return this._errorEvent;
+        }
+    },
+
+
+    ready : {
+        get : function() {
+            return this._ready;
+        }
+    },
+
+
+    hasAlphaChannel : {
+        get : function() {
+            return true;
+        }
+    }
+});
+
+MapboxVectorTileImageryProvider.prototype._getSubdomain = function(x, y, level) {
+    if (this._subdomains.length === 0) {
+        return undefined;
+    }
+    else {
+        var index = (x + y + level) % this._subdomains.length;
+        return this._subdomains[index];
+    }
+};
+
+MapboxVectorTileImageryProvider.prototype._buildImageUrl = function(x, y, level) {
+    return this._uriTemplate.expand({
+        z: level,
+        x: x,
+        y: y,
+        s: this._getSubdomain(x, y, level),
+    });
+};
+
+
+/*MapboxVectorTileImageryProvider.prototype.getTileCredits = function(x, y, level) {
+    return undefined;
+};*/
+
+
+MapboxVectorTileImageryProvider.prototype._requestImage = function(x, y, level, canvas) {
+    var that = this;
+    var url = this._buildImageUrl(x, y, level);
+
+    return loadArrayBuffer(url).then(function(data) {
+        var tile = new VectorTile(new Protobuf(data));
+        var layer = tile.layers[that._layerName];
+
+        if (!defined(layer)) {
+            return canvas; // return blank canvas for blank tile
+        }
+
+        var context = canvas.getContext('2d');
+        context.strokeStyle = "black";
+        context.lineWidth = 1;
+
+        var pos;
+
+        var extentFactor = canvas.width/layer.extent; // Vector tile works with extent [0, 4095], but canvas is only [0,255]
+
+        for (var i = 0; i < layer.length; i++) {
+            var feature = layer.feature(i);
+            if (feature.type === POLYGON_FEATURE) {
+                var style = that._styleFunc(feature.properties[that._featurePicking.uniqueIdProp]);
```

Comment:
Since `_styleFunc` appears to be required, it would be good if the constructor checked to make sure it's provided.